iothub_client/src/iothubtransport_mqtt_common.c

```diff
@@ -1878,6 +1878,7 @@ static void mqtt_error_callback(MQTT_CLIENT_HANDLE handle, MQTT_CLIENT_EVENT_ERR
             case MQTT_CLIENT_NO_PING_RESPONSE:
             {
                 LogError("Mqtt Ping Response was not encountered.  Reconnecting device...");
+                transport_data->transport_callbacks.connection_status_cb(IOTHUB_CLIENT_CONNECTION_UNAUTHENTICATED, IOTHUB_CLIENT_CONNECTION_COMMUNICATION_ERROR, transport_data->transport_ctx);
```

Comment:
>IOTHUB_CLIENT_CONNECTION_COMMUNICATION_ERROR [](start = 115, length = 44)

Should this be a new Status reason?  Maybe No Ping response?

Reply:
this seems reasonable. totally up to you

---
In reply to: [322858359](https://github.com/Azure/azure-iot-sdk-c/pull/1161#discussion_r322858359) [](ancestors = 322858359)